lld/wasm/Driver.cpp

```diff
@@ -965,6 +965,8 @@ static void createSyntheticSymbols() {
   } else {
     // For non-PIC code
     WasmSym::stackPointer = createGlobalVariable("__stack_pointer", true);
+    WasmSym::definedMemoryBase = symtab->addOptionalDataSymbol("__memory_base");
+    WasmSym::definedTableBase = symtab->addOptionalDataSymbol("__table_base");
```

Comment:
these calls seem too early in the link process.
the symbol table is almost empty here and they are effectively no-op i guess.
createSyntheticSymbols is called even before addFile.

Reply:
ok.
i will try to refresh my memory first. :-)

Reply:
see https://github.com/llvm/llvm-project/pull/153537